java/arcs/core/host/AbstractArcHost.kt

```diff
@@ -0,0 +1,33 @@
+/*
+ * Copyright 2019 Google LLC.
+ *
+ * This code may only be used under the BSD style license found at
+ * http://polymer.github.io/LICENSE.txt
+ *
+ * Code distributed by Google as part of this project is also subject to an additional IP rights
+ * grant found at
+ * http://polymer.github.io/PATENTS.txt
+ */
+package arcs.core.host
+
+import arcs.core.sdk.Particle
+
+/**
+ * Base helper class for [ArcHost] implementations to provide implementation of
```

Comment:
Any reason to have both an `ArcHost` interface and an `AbstractArcHost` abstract class? We could just make `ArcHost` an abstract base class.

Reply:
The original intent is that there are some Services in AiAi which already have a base class, and the idea was to allow them to also be turned into ArcHost's by implementing the interface. Another reason is that we should support ArcHosts which don't allow other people to register particles in them.

So for example, if I make an AiAiAutofillArcHost, I should bake-in my registerd ArcsAutofillParticle, but not allow someone else to mutate the results of availableParticles().  Although this suggests I should remove the registerParticle()/unregisterParticle() from the interface, or document that it can throw 'unsupported operation'